### DISCUSSION TITLE
How to reboot a single core in polarfire soc FPGA

### DISCUSSION THREAD

**Discussion by ZGDIGE** [polarfire-soc/polarfire-soc-documentation#495] [HSS and U-Boot]
How can I reboot a core without affecting the normal operation of other cores?Are there any official examples?

**Comment by hughbreslin** [comment]
Hey @ZGDIGE have you seen the [reboot documentation](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/hss-and-u-boot/reboot.md#overview)?

**Reply by ZGDIGE** [reply]
Yes, I saw this document and I want to know if it is possible to do it without PMP.

**Reply by hughbreslin** [reply]
As in use the reboot functionality without configuring any PMPs?

**Reply by ZGDIGE** [reply]
Yes, after I used the ip core of FDMA, there was a conflict with the PMP of MSS, so I disabled PMP.

**Reply by hughbreslin** [reply]
Theres no requirement to use this feature with the PMPs, they're independent. If you want to use the PMPs one is automatically reserved for openSBI.

**Reply by ZGDIGE** [reply]
OK, thank you. Do I just need to follow the reboot documentation?

**Reply by hughbreslin** [reply]
Pretty much :) If you have a look at our AMP demos we don't have PMPs enabled in the base reference designs but the reboot functionality is there

**Reply by ZGDIGE** [reply]
Thank you so much, I'll try.

**Reply by ZGDIGE** [reply]
@hughbreslin Where can I get the AMP demo?

**Reply by hughbreslin** [reply]
The AMP demos are included as part of Yocto and Buildroot targeting the Icicle Kit, theres some documentation [here](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/applications-and-demos/asymmetric-multiprocessing/amp.md#building-the-linux--freertosbare-metal-demo) and you can also see AMP listed as a machine in Yocto (along with pre built executables provided with the releases)

**Reply by ZGDIGE** [reply]
Thank you.

**Reply by ZGDIGE** [reply]
@hughbreslin The 2021 version of HSS I am currently using does not have the mpfs_hart_stop function. Does this mean that the 2021 version of HSS does not support single-core restart?

**Reply by hughbreslin** [reply]
It looks like the feature was added in this [commit](https://github.com/polarfire-soc/hart-software-services/commit/833048036c2837503656dc53e251a9d017dda055) from September 2022 so no it wouldn't be in the 2021 version, you'll need to update to the latest version for this feature

**Reply by ZGDIGE** [reply]
Thank you.
